spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -272,7 +272,7 @@ private void restart() {
 	protected void doStop() {
 		try {
 			this.active = false;
-			this.lifecycleCondition.await(Math.min(this.stopTimeout, this.receiveTimeout), TimeUnit.MICROSECONDS);
+			this.lifecycleCondition.await(Math.min(this.stopTimeout, this.receiveTimeout), TimeUnit.MILLISECONDS);
```

Comment:
Yes; I mean if they increase the timeouts to 10s and have 10 adapters, it could take up to 100 seconds to stop the context because they are stopped serially :smile: